test/shared/utils/text_input_util_test.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:didpay/shared/utils/text_input_util.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('TextInputUtil', () {
+    test('should give phone keyboard type', () {
+      const pattern = r'^\+2547[0-9]{8}$';
+
+      expect(TextInputUtil.getKeyboardType(pattern), TextInputType.phone);
+    });
+
+    test('should give text keyboard type', () {
+      const pattern = r'[a-zA-Z]';
+
+      expect(TextInputUtil.getKeyboardType(null), TextInputType.text);
+      expect(TextInputUtil.getKeyboardType(pattern), TextInputType.text);
+    });
+
+    test('should give US phone number mask', () {
+      const pattern = r'^\+1[0-9]{10}$';
+      final formatter = TextInputUtil.getMaskFormatter(pattern);
+
+      expect(formatter.getMask(), '+1 (XXX) XXX-XXXX');
+    });
+
+    test('should give Mexico phone number mask', () {
+      const pattern = r'^\+52[0-9]{10}$';
+      final formatter = TextInputUtil.getMaskFormatter(pattern);
+
+      expect(formatter.getMask(), '+52 XX XXXX-XXXX');
+    });
+
+    test('should give Ghana phone number mask', () {
+      const pattern = r'^\+233[0-9]{9}$';
+      final formatter = TextInputUtil.getMaskFormatter(pattern);
+
+      expect(formatter.getMask(), '+233 2XX XXXXXX');
+    });
+
+    test('should give Kenya phone number mask', () {
+      const pattern = r'^\+2547[0-9]{8}$';
+      final formatter = TextInputUtil.getMaskFormatter(pattern);
+
+      expect(formatter.getMask(), '+254 7XX XXXXXX');
+    });
```

Comment:
Given that this is PFI agnostic, will we have to maintain this list for all new countries we plan to support

Reply:
@wesbillman yea pretty much.. wanted to create the masks dynamically but every country has their own specific format. given that we have a list of supported countries in the onboarding flow, maybe we can update the country-specific masks as more supported countries are added.

@kirahsapong totally! great point about some people having phone numbers from other countries. yellow card actually enforced country-specific phone numbers based on payout method so that's the inspiration for this pr! the country-specific input mask only comes to effect if the pfi includes a pattern that asks for a supported country calling code so the default is no input mask (or one that only enforces length if something like `^[0-9]{10}` is the pattern).
and this pr is actually for the input masking in `PaymentDetailsPage` when the user is inputting their account number, phone number, etc!

Reply:
Oh I wonder if we can add a mask to the data here? https://github.com/TBD54566975/pfi-providers-data/blob/main/pfis.json

Assuming something like this will be our "source of truth" we might be able to utilize it for config stuff as well.